blind_walking/net/feature_encoder.py

```diff
@@ -21,22 +30,61 @@ def __init__(
         self,
         observation_space: gym.spaces.Dict,
         mlp_output_size: int = 8,
+        cnn_output_size: int = 8,
     ):
-        self.mlp_output_size = mlp_output_size
-        flatten_output_size = math.prod(observation_space.spaces["flatten"].shape)
-        assert len(observation_space.spaces["mlp"].shape) == 1, "Mlp-encoded space must be 1-dimensional"
-        mlp_input_size = observation_space.spaces["mlp"].shape[0]
-        features_dim = flatten_output_size + mlp_output_size
+        features_dim = 0
+
+        self.flatten_output_size = math.prod(observation_space.spaces["flatten"].shape)
+        features_dim += self.flatten_output_size
+
+        self._contains_mlp = contains_subspace(observation_space, "mlp")
+        self._contains_cnn = contains_subspace(observation_space, "LocalTerrainView_cnn")
```

Comment:
visual encoder is not used together with feature encoder as of now. note that the goal is NOT to reproduce the rma paper, but to have our own contribution which is to have all-terrain traversibility (probably through visual perception or smth)